pqdb-frontend/src/components/views.js

```diff
@@ -1,53 +1,273 @@
+                        <Typography paragraph>
+                            Contributions are warmly welcomed, see <Link href="https://github.com/cryptoeng/pqdb#contribute">here</Link> for details.
+                        </Typography>
+
+                        <Typography variant='h5' align='center'>Available Schemes</Typography>
+                        <Box mt={1} />
+                        <Grid container justify="space-evenly">
+                            <Grid item>
+                                <Paper>
+                                    <Box p={4}>
+                                        <Typography variant="h6">Encryption Schemes</Typography>
+                                        <List>
+                                            {this.state.enc.map(name => <ListItem key={name}><ListItemText>{name}</ListItemText></ListItem>)}
+                                        </List>
+                                    </Box>
+                                </Paper>
+                            </Grid>
+                            <Grid item>
+                                <Paper>
+                                    <Box p={4}>
+                                        <Typography variant="h6">Signature Schemes</Typography>
+                                        <List>
+                                            {this.state.sig.map(name => <ListItem key={name}><ListItemText>{name}</ListItemText></ListItem>)}
+                                        </List>
+                                    </Box>
+                                </Paper>
+                            </Grid>
+                        </Grid>
+                    </Box>
+                </Paper>
+            </Container>
+        );
+    }
 }
 
 class CustomSQLQuery extends React.Component {
     constructor(props) {
         super(props);
         this.db = props.db;
+        this.params = qs.parse(this.props.history.location.search);
+        var sqlInput = ('query' in this.params) ? this.params['query'] : '';
+
         this.state = {
-            sqlQuery: "",
+            sqlInput: sqlInput,
+            executedSqlQuery: '',
             queryResult: null,
             error: null,
         }
     }
 
+    componentDidMount() {
+        if (this.state.sqlInput !== '') this.executeSQLQuery();
+    }
+
     executeSQLQuery() {
+        const history = this.props.history;
+        this.params.query = this.state.sqlInput;
+        history.replace({
+            pathname: history.location.pathname,
+            search: '?' + qs.stringify(this.params)
+        });
+
+        var sqlQuery = this.state.sqlInput;
+
         try {
-            var results = this.db.exec(this.state.sqlQuery)[0];
-            console.log(results);
-            this.setState({ queryResult: results, error: null });
-        } catch (e) {
-            this.setState({ queryResult: null, error: e });
+            var results = this.db.exec(sqlQuery)[0];
```

Comment:
we should put some indicator that the query is running. complex queries might take a while. we could make the mouse pointer a wheel while this is executing, or replace the "run query" by "cancel" or grey it out or something or all of it

Reply:
Yep, my test queries weren't that complex, Going to add some indicator.